<h1>ТИПиС. ИДЗ №1. Лохматикн К.Д. гр. №3391</h1>

In [ ]:
pip install ucimlrepo # Ставим либу для подкачки данных

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 25.1.1 -> 25.2
[notice] To update, run: python.exe -m pip install --upgrade pip


In [ ]:
import pandas as pd # Импортируем пандас

In [ ]:
# Загружаем датасет с помощью готового куска кода с сайта, чутка модифицировав

from ucimlrepo import fetch_ucirepo

# fetch dataset 
adult = fetch_ucirepo(id=2) 
  
# data (as pandas dataframes) 
X = pd.concat([adult.data.features, adult.data.targets], axis = 1)
X

age         workclass  fnlwgt  education  education-num  \
0       39         State-gov   77516  Bachelors             13   
1       50  Self-emp-not-inc   83311  Bachelors             13   
2       38           Private  215646    HS-grad              9   
3       53           Private  234721       11th              7   
4       28           Private  338409  Bachelors             13   
...    ...               ...     ...        ...            ...   
48837   39           Private  215419  Bachelors             13   
48838   64               NaN  321403    HS-grad              9   
48839   38           Private  374983  Bachelors             13   
48840   44           Private   83891  Bachelors             13   
48841   35      Self-emp-inc  182148  Bachelors             13   

           marital-status         occupation    relationship  \
0           Never-married       Adm-clerical   Not-in-family   
1      Married-civ-spouse    Exec-managerial         Husband   
2                Divorced  Handlers-cleaners   Not-in-family   
3      Married-civ-spouse  Handlers-cleaners         Husband   
4      Married-civ-spouse     Prof-specialty            Wife   
...                   ...                ...             ...   
48837            Divorced     Prof-specialty   Not-in-family   
48838             Widowed                NaN  Other-relative   
48839  Married-civ-spouse     Prof-specialty         Husband   
48840            Divorced       Adm-clerical       Own-child   
48841  Married-civ-spouse    Exec-managerial         Husband   

                     race     sex  capital-gain  capital-loss  hours-per-week  \
0                   White    Male          2174             0              40   
1                   White    Male             0             0              13   
2                   White    Male             0             0              40   
3                   Black    Male             0             0              40   
4                   Black  Female             0             0              40   
...                   ...     ...           ...           ...             ...   
48837               White  Female             0             0              36   
48838               Black    Male             0             0              40   
48839               White    Male             0             0              50   
48840  Asian-Pac-Islander    Male          5455             0              40   
48841               White    Male             0             0              60   

      native-country  income  
0      United-States   <=50K  
1      United-States   <=50K  
2      United-States   <=50K  
3      United-States   <=50K  
4               Cuba   <=50K  
...              ...     ...  
48837  United-States  <=50K.  
48838  United-States  <=50K.  
48839  United-States  <=50K.  
48840  United-States  <=50K.  
48841  United-States   >50K.  

[48842 rows x 15 columns]

In [ ]:
len(X.columns) # Кол-во столбцов в таблице

15

In [ ]:
len(X.index) # Кол-во объектов в таблице

48842

In [ ]:
X.info() # Информация по пропускам

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             48842 non-null  int64 
 1   workclass       47879 non-null  object
 2   fnlwgt          48842 non-null  int64 
 3   education       48842 non-null  object
 4   education-num   48842 non-null  int64 
 5   marital-status  48842 non-null  object
 6   occupation      47876 non-null  object
 7   relationship    48842 non-null  object
 8   race            48842 non-null  object
 9   sex             48842 non-null  object
 10  capital-gain    48842 non-null  int64 
 11  capital-loss    48842 non-null  int64 
 12  hours-per-week  48842 non-null  int64 
 13  native-country  48568 non-null  object
 14  income          48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


In [ ]:
len(X['race'].unique()) # Уникальные значения в столбце

5

In [ ]:
X['hours-per-week'].median() # Медиана столбца

40.0

In [ ]:
X.loc[(X['sex'] == 'Male') & (X['income'] == '>50K')].shape[0] # Выделяем часть датасета по двум условиям, находим размерность и ищем кол-во объектов

6662

In [ ]:
X.loc[(X['sex'] == 'Female') & (X['income'] == '>50K')].shape[0] # Аналогично предыдущей строчке

1179

In [ ]:
# Заполняем пропуски модами этих же столбцов и проверяем что пропусков больше в таблице нет
X['workclass'].fillna(X['workclass'].mode()[0], inplace = True)
X['occupation'].fillna(X['occupation'].mode()[0], inplace = True)
X['native-country'].fillna(X['native-country'].mode()[0], inplace = True)
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             48842 non-null  int64 
 1   workclass       48842 non-null  object
 2   fnlwgt          48842 non-null  int64 
 3   education       48842 non-null  object
 4   education-num   48842 non-null  int64 
 5   marital-status  48842 non-null  object
 6   occupation      48842 non-null  object
 7   relationship    48842 non-null  object
 8   race            48842 non-null  object
 9   sex             48842 non-null  object
 10  capital-gain    48842 non-null  int64 
 11  capital-loss    48842 non-null  int64 
 12  hours-per-week  48842 non-null  int64 
 13  native-country  48842 non-null  object
 14  income          48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


<h2>Ответы на вопросы.</h2>
1. Всего столбцов в датасете 15.<br>
2. Пропуски данных в датасете имеются, конкретно в столбцах workclass, occupation и native-country.<br>
3. Уникальных значений расы равно 5.<br>
4. Медиана house-per-week равна 40.<br>
5. В датасете больше мужчин, которые зарабатывают больше 50K (6662 против 1179).<br>
6. Пропуски были заполнены модами столбцов. Числовые пропуски можно было бы заменить на ноль, среднее или медиану, категориальные можно было бы заполнить новой категорией вроде 'Missing'.<br>